src/main/kotlin/com/autonomousapps/subplugin/ProjectPlugin.kt

```diff
@@ -692,6 +692,25 @@ internal class ProjectPlugin(private val project: Project) {
     }
   }
 
+  private fun Project.supportedSourceSetNames() = provider {
```

Comment:
When I checked the explicit type of this, I got `Provider<Collection<String>>`, when I would have expected `Provider<Set<String>>` (from line 710). I think we can achieve that (which I prefer), with this small change:
```kotlin
the<AppExtension>().applicationVariants.flatMapToSet {
  it.sourceSets.map { sourceSet -> sourceSet.name } +
    (it.unitTestVariant?.sourceSets?.map { sourceSet -> sourceSet.name } ?: emptySet())
  // Not yet supported: + it.testVariant.sourceSets.map { sourceSet -> sourceSet.name }
}
```
Note the use of the `flatMapToSet` extension function.

(the same change works for the `LibraryExtension` section.

Reply:
And then as a style matter, could you move this whole function _below_ the one that uses it? Semantically, I see this as more of a helper function, and therefore less important than the `configureAggregationTasks()` function.

Reply:
Yes that's better. I made the changes and also added a short Javadoc comment to the function.